AfterClass-io/afterclass.io-v2
# chore: hide search result item statistics on unauthenticated client

~blocked by #191, to be reviewed after #191 merges~

## Context


Following up to the changes introduced in #191, the count procedure is public and will allow anyone with knowledge of the endpiont to query for count of all reviews, which is undesirable

This PR makes count a private trpc procedure and removes review count and prof/course count on search results (previously relying on public procedure to display)

## Changes

- make `api.reviews.count` a private trpc procedure
- minor bugfixes to promise.all

## How to Test


1. head to site and don't authenticate
2. do search
3. no longer see stats on search results

## Preview / Screenshots



| before | after |
|:-:|:-:|
|  ![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/82136017-44a7-4410-ab5c-1e13d0a3378f) | ![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/b92af5b6-a6ab-4189-b576-c40de068ed5c) |

## Checklist
